ouroboros-consensus/src/Ouroboros/Consensus/Util.hs

```diff
@@ -346,6 +350,10 @@ allDisjoint = go Set.empty
     go _   []       = True
     go acc (xs:xss) = Set.disjoint acc xs && go (Set.union acc xs) xss
 
+-- | Map over keys and values
```

Comment:
Why is this called a `dimap` and not a `bimap`? My guess: `dimap` does not satisfy the bifunctor laws (I haven't checked).

Reply:
It is a profunctor I think, in the sense that it contravariant in the first argument, covariant in the latter. The profunctor map in https://hackage.haskell.org/package/profunctors-5.6.2/docs/Data-Profunctor.html is called `dimap`.